pkg/registry/core/service/ipallocator/controller/repairip.go

```diff
@@ -0,0 +1,539 @@
+		r.svcQueue.AddRateLimited(key)
+		return
+	}
+
+	klog.Warningf("Dropping Service %q out of the queue: %v", key, err)
+	r.svcQueue.Forget(key)
+	runtime.HandleError(err)
+}
+
+// syncServices reconcile the Service ClusterIPs to verify that each one has the corresponding IPAddress object associated
+func (r *RepairIPAddress) syncService(key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	svc, err := r.serviceLister.Services(namespace).Get(name)
+	if err != nil {
+		// nothing to do
+		return nil
+	}
+	if !helper.IsServiceIPSet(svc) {
+		// didn't need a ClusterIP
+		return nil
+	}
+
+	for _, clusterIP := range svc.Spec.ClusterIPs {
+		ip := netutils.ParseIPSloppy(clusterIP)
+		if ip == nil {
+			// ClusterIP is corrupt, ClusterIPs are already validated, but double checking here
+			// in case there are some inconsistencies with the parsers
+			r.recorder.Eventf(svc, nil, v1.EventTypeWarning, "ClusterIPNotValid", "ClusterIPValidation", "Cluster IP %s is not a valid IP; please recreate Service", ip)
+			runtime.HandleError(fmt.Errorf("the ClusterIP %s for Service %s/%s is not a valid IP; please recreate Service", ip, svc.Name, svc.Namespace))
+			continue
+		}
+
+		family := getFamilyByIP(ip)
+		if network, ok := r.networkByFamily[family]; !ok {
+			// this service is using an IPFamily no longer configured on cluster
+			r.recorder.Eventf(svc, nil, v1.EventTypeWarning, "ClusterIPNotValid", "ClusterIPValidation", "Cluster IP %s(%s) is of ip family that is no longer configured on cluster; please recreate Service", ip, family)
+			runtime.HandleError(fmt.Errorf("the ClusterIP [%v]: %s for Service %s/%s is of ip family that is no longer configured on cluster; please recreate Service", family, ip, svc.Name, svc.Namespace))
+			continue
+		} else if !network.Contains(ip) {
+			// ClusterIP is out of range
+			r.recorder.Eventf(svc, nil, v1.EventTypeWarning, "ClusterIPOutOfRange", "ClusterIPAllocation", "Cluster IP [%v]: %s is not within the configured Service CIDR %s; please recreate service", family, ip, network.String())
+			runtime.HandleError(fmt.Errorf("the ClusterIP [%v]: %s for Service %s/%s is not within the service CIDR %s; please recreate", family, ip, svc.Name, svc.Namespace, network.String()))
+			continue
+		}
+
+		ipaddress, err := r.ipAddressLister.Get(ip.String())
+		if apierrors.IsNotFound(err) {
+			// ClusterIP doesn't seem to be allocated,
+			r.recorder.Eventf(svc, nil, v1.EventTypeWarning, "ClusterIPNotAllocated", "ClusterIPAllocation", "Cluster IP [%v]: %s is not allocated; repairing", family, ip)
+			runtime.HandleError(fmt.Errorf("the ClusterIP [%v]: %s for Service %s/%s is not allocated; repairing", family, ip, svc.Name, svc.Namespace))
+			_, err := r.client.NetworkingV1alpha1().IPAddresses().Create(context.Background(), newIPAddress(ip.String(), family, svc), metav1.CreateOptions{})
+			if err != nil {
+				return err
+			}
+			continue
+		}
+		if err != nil {
```

Comment:
move this right after `ipaddress, err := r.ipAddressLister.Get(ip.String())`
Get the result 
if error && !IsNotFound (bail out, can't do anything now really)
if err && isNotFound(create or fail)
if no error && not owned by the service (own it?)